src/pages/index.js

```diff
@@ -104,35 +129,104 @@ function closeModalByEscape(event) {
+const saveButtonAvatar = avatarForm.querySelector(".modal__submit-btn");
+
+// Function to reset button text and state
+function resetSaveButton(saveButton) {
+  saveButton.textContent = "Save";
+  saveButton.disabled = false;
+  saveButton.classList.remove("modal__submit-btn_disabled");
+}
+
 // Handle Edit Form Submit
 function handleEditFormSubmit(evt) {
   evt.preventDefault();
-  profileName.textContent = profileNameInput.value;
-  profileDescription.textContent = profileDescriptionInput.value;
-  closeModal(editModal);
+  saveButtonEdit.textContent = "Saving...";
+
+  api
+    .editUserInfo({
+      name: profileNameInput.value,
+      about: profileDescriptionInput.value,
+    })
+    .then((data) => {
+      profileName.textContent = data.name;
+      profileDescription.textContent = data.about;
+      closeModal(editModal);
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      resetSaveButton(saveButtonEdit);
+    });
 }
 
 // Handle Add Card Submit
 function handleAddCardSubmit(evt) {
   evt.preventDefault();
-  const newCard = getCardElement({
+
+  saveButtonCard.textContent = "Saving...";
+
+  const cardData = {
     name: cardNameInput.value,
     link: cardLinkInput.value,
-  });
-  cardsList.prepend(newCard);
-  closeModal(cardModal);
-  cardForm.reset();
+  };
+
+  api
+    .addCard(cardData)
+    .then((newCard) => {
+      const cardElement = getCardElement(newCard);
+      cardsList.prepend(cardElement);
+      cardForm.reset();
+      closeModal(cardModal);
+
+      toggleButtonState(
+        Array.from(cardForm.querySelectorAll(validationConfig.inputSelector)),
+        saveButtonCard,
+        validationConfig
+      );
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      resetSaveButton(saveButtonCard);
+    });
+}
 
-  toggleButtonState(
-    Array.from(cardForm.querySelectorAll(validationConfig.inputSelector)),
-    cardForm.querySelector(validationConfig.submitButtonSelector),
-    validationConfig
-  );
+// Avatar Handler
+function handleAvatarSubmit(evt) {
+  evt.preventDefault();
+
+  saveButtonAvatar.textContent = "Saving...";
+  saveButtonAvatar.disabled = true;
+  saveButtonAvatar.classList.add("modal__submit-btn_disabled");
+
+  api
+    .editAvatarInfo(avatarInput.value)
+    .then((data) => {
+      profileAvatar.src = data.avatar;
+      closeModal(avatarModal);
+      avatarForm.reset();
+    })
+    .catch(console.error)
+    .finally(() => {
+      resetSaveButton(saveButtonAvatar);
+    });
 }
 
 // Form Submission Handlers
 editForm.addEventListener("submit", handleEditFormSubmit);
 cardForm.addEventListener("submit", handleAddCardSubmit);
+avatarForm.addEventListener("submit", handleAvatarSubmit);
+
+// Form Submission Handlers
+editForm.addEventListener("submit", handleEditFormSubmit);
+cardForm.addEventListener("submit", handleAddCardSubmit);
+avatarForm.addEventListener("submit", handleAvatarSubmit);
```

Comment:
Please, delete the 3 duplicated submit handlers because you have them in lines 222-224